app/services/delivery_request_service.rb

```diff
@@ -31,13 +31,21 @@ def call
 
     status = Metrics.email_send_request(provider_name) { send_email }
 
-    return attempt if status == :sent
+    if status == :sent
```

Comment:
Could we simplify this code a bit? It's confusing that we're:

- Doing some of the updates in a transaction.
- Sometimes checking `attempt` vs. `status`.
- Implicitly handling the other (comms) status.

We could have something like:

```
ActiveRecord::Base.transaction do
  case status
  when :sent ...
  when :delivered ...
  when :undeliverable_failure ...
  when :provider_communication_failure ...
end
```

Reply:
Although this may introduce a little duplication, I think it's worth it for the clarity.

Reply:
Sure thing, I've rejigged this bit around and added some missing tests around recording metrics. I did actually start writing a case statement but then changed my mind, have added said case.